projects/example-app/src/app/reducers/index.ts

```diff
@@ -16,25 +17,28 @@ import * as fromRouter from '@ngrx/router-store';
  */
 
 import * as fromLayout from '@example-app/core/reducers/layout.reducer';
+import { SerializedRouterStateSnapshot } from '@ngrx/router-store';
 
 /**
  * As mentioned, we treat each reducer like a table in a database. This means
  * our top level state interface is just a map of keys to inner state types.
  */
 export interface State {
   layout: fromLayout.State;
-  router: fromRouter.RouterReducerState;
+  router: fromRouter.RouterReducerState<any>;
 }
 
 /**
  * Our state is composed of a map of action reducer functions.
  * These reducer functions are called with each dispatched action
  * and the current or initial state and return a new immutable state.
  */
-export const reducers: ActionReducerMap<State> = {
-  layout: fromLayout.reducer,
-  router: fromRouter.routerReducer,
-};
+export function reducers(state: State | undefined, action: Action) {
```

Comment:
This isn't valid for the `StoreModule.forRoot()` signature that expects `ActionReducerMap<State, Action> | InjectionToken<ActionReducerMap<State, Action>>`. We're currently using the map of reducers and `combineReducers` to create the root reducer internally. 

We'll need some additional tests to verify this works with `StoreModule.forRoot()` also.

Reply:
You are right, it failed to start
```
ERROR in projects/example-app/src/app/app.module.ts(36,25): error TS2345: Argument of type '(state: State | undefined, action: Action) => { layout: State; router: RouterReducerState<BaseRouterStoreState>; }' is not assignable to parameter of type 'ActionReducerMap<State, Action> | InjectionToken<ActionReducerMap<State, Action>>'.
  Type '(state: State | undefined, action: Action) => { layout: State; router: RouterReducerState<BaseRouterStoreState>; }' is missing the following properties from type 'InjectionToken<ActionReducerMap<State, Action>>': _desc, ngInjectableDef
```

I'll see if i can make it take a function as well.

Reply:
Working on it in another PR